data/account/participation.go

```diff
@@ -217,12 +217,9 @@ func FillDBWithParticipationKeys(store db.Accessor, address basics.Address, firs
 	}
 	// Persist the Participation into the database
 	err = part.Persist()
-	if err != nil {
-		return PersistedParticipation{}, err
+	if err == nil {
+		err = stateProofSecrets.Persist() // must be called after part.Persist()
```

Comment:
If Persist() function is not written well, you will return a PersistedParticipation struct which might be filled with wrong data. I think, In order to protect ourselves, we should return empty struct.

for the err value- this change is great.

Reply:
The expectations should be that if error is returned, there are no guarantees on anything else returned (this was something I fixed on master).

When you say "protect ourselves", what are the risks? Maybe, we should make sure Persist is safe when it returns an error.

I have no problem with returning PersistedParticipation{} in case of an error. However, if there are risks with returning part, we need to understand that,